views/js/runner/plugins/navigation/nextWarningHelper.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * This is a helper for navigation plugins. It decides if a warning should be displayed
+ * before actually moving to the next item
+ *
+ * @author Christophe Noël <christophe@taotesting.com>
+ */
+define([], function () {
+    'use strict';
+
+    /**
+     * @param {Object} options
+     * @param {Boolean} options.endTestWarning - enables the end test warning, when applicable
+     * @param {Boolean} options.isLast - if the item is the last of the test
+     * @param {Boolean} options.isLinear - if the current part is linear
+     * @param {Boolean} options.nextItemWarning - enables the next item warning, when applicable
+     * @param {Object} options.nextPart - description of the next part of the test
+     * @param {Number} options.remainingAttempts - remaining attempts for the current item
+     * @param {String} options.testPartId - current test part identifier
+     * @returns {Object}
+     */
+    var nextWarningHelper = function nextWarningHelper(options) {
+        var endTestWarning      = toBoolean(options.endTestWarning, false),
+            isLast              = toBoolean(options.isLast, false),
+            isLinear            = toBoolean(options.isLinear, false),
+            nextItemWarning     = toBoolean(options.nextItemWarning, false),
+            nextPart            = options.nextPart || {},
+            remainingAttempts   = typeof(options.remainingAttempts) === 'undefined' ? -1 : options.remainingAttempts,
+            testPartId          = options.testPartId || '',
+
+            warnBeforeNext = shouldWarnBeforeNext(),
+            warnBeforeEnd = shouldWarnBeforeEnd();
+
+        /**
+         * Convert a value to a boolean
+         * @param {*} value
+         * @param {Boolean} defaultValue
+         * @returns {Boolean}
+         */
+        function toBoolean(value, defaultValue) {
```

Comment:
This function could be external to its current scope, as there is nothing related to a local variable. Moreover this would ease the read as this function is used in var declarations